src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package net.oauth.jsontoken;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.AsyncFunction;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gson.JsonObject;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import net.oauth.jsontoken.crypto.SignatureAlgorithm;
+import net.oauth.jsontoken.crypto.Verifier;
+import net.oauth.jsontoken.discovery.AsyncVerifierProvider;
+import net.oauth.jsontoken.discovery.AsyncVerifierProviders;
+
+/**
+ * The asynchronous counterpart of {@link JsonTokenParser}.
+ * Class that parses and verifies JSON Tokens asynchronously.
+ */
+public final class AsyncJsonTokenParser extends AbstractJsonTokenParser {
+  private final AsyncVerifierProviders asyncVerifierProviders;
+  private final Executor executor;
+
+  /**
+   * Creates a new {@link AsyncJsonTokenParser}.
+   *
+   * @param clock a clock object that will decide whether a given token is currently valid or not.
+   * @param asyncVerifierProviders an object that provides signature verifiers asynchronously
+   *   based on a signature algorithm, the signer, and key ids.
+   * @param executor an executor to run the tasks before and after getting the verifiers
+   * @param checkers an array of checkers that validates the parameters in the JSON token.
+   */
+  public AsyncJsonTokenParser(
+      Clock clock, AsyncVerifierProviders asyncVerifierProviders, Executor executor, Checker... checkers) {
```

Comment:
nit: 100 character limit on line length: https://google.github.io/styleguide/javaguide.html#s4.4-column-limit

Reply:
Fixed here and in other lines.